### PR TITLE
[ESP32] Enable extended ble announcement for esp32 platform

### PR DESCRIPTION
**Pull request by sayondeep**
Extended BLE advertisement was recently introduced in this [PR](https://github.com/project-chip/connectedhomeip/pull/31532).
This PR enables extended BLE advertisement for ESP32 platform.
The extended BLE advertisement increases the advertisement duration to up to 48 hr before closing the commissioning window.

**Tests Performed:**

- Tested successful commissioning at all three phases (Fast advertisement, Slow advertisement and Extended advertisement).

### CODE REVIEW

File: config/esp32/components/chip/Kconfig [hunk 1]
```diff
@@ -1200,4 +1200,13 @@ menu "CHIP Device Layer"
 
     endmenu
 
+    menu "Enable BLE Extended Advertisement"
```
Comment (dhrishi) [inline config/esp32/components/chip/Kconfig:1203]: extended adv is a standard feature of BLE 5.0
Can we change this to -

BLE extended announcement

File: config/esp32/components/chip/Kconfig [hunk 2]
```diff
@@ -1200,4 +1200,13 @@ menu "CHIP Device Layer"
...
+        config ENABLE_BLE_EXT_ADVERTISING
+            bool "Enable BLE Extended Advertisement"
```
Comment (dhrishi) [inline config/esp32/components/chip/Kconfig:1205]: Enable BLE extended announcement

File: config/esp32/components/chip/Kconfig [hunk 3]
```diff
@@ -1200,4 +1200,13 @@ menu "CHIP Device Layer"
...
+            default n
+            help
+                Enable BLE Extended Advertisement.
```
Comment (dhrishi) [inline config/esp32/components/chip/Kconfig:1208]: Enable BLE extended announcement

File: src/platform/ESP32/BLEManagerImpl.h [hunk 4]
```diff
@@ -232,6 +232,7 @@ class BLEManagerImpl final : public BLEManager,
         kFastAdvertisingEnabled   = 0x0200, /**< The application has enabled fast advertising. */
         kUseCustomDeviceName      = 0x0400, /**< The application has configured a custom BLE device name. */
         kAdvertisingRefreshNeeded = 0x0800, /**< The advertising configuration/state in ESP BLE layer needs to be updated. */
+        kExtAdvertisingEnabled    = 0x1000, /**< The application has enabled Extended BLE advertising. */
```
Comment (dhrishi) [inline src/platform/ESP32/BLEManagerImpl.h:235]: has enabled extended announcement

File: src/platform/ESP32/nimble/BLEManagerImpl.cpp [hunk 5]
```diff
@@ -76,6 +77,9 @@ namespace Internal {
 
 namespace {
 
+TimerHandle_t sbleAdvTimeoutTimer; // FreeRTOS sw timer.
+
+
```
Comment (dhrishi) [inline src/platform/ESP32/nimble/BLEManagerImpl.cpp:82]: Remove a blank line?

File: src/platform/ESP32/nimble/BLEManagerImpl.cpp [hunk 6]
```diff
@@ -221,6 +223,15 @@ CHIP_ERROR BLEManagerImpl::_Init()
 #endif
     SuccessOrExit(err);
 
+    // Create FreeRTOS sw timer for BLE timeouts and interval change.
+    sbleAdvTimeoutTimer = xTimerCreate("BleAdvTimer",       // Just a text name, not used by the RTOS kernel
+                                    //    pdMS_TO_TICKS(50),
+                                       1,   // == default timer period
+                                       false,               // no timer reload (==one-shot)
+                                       (void *) this,       // init timer id = ble obj context
```
Comment (dhrishi) [inline src/platform/ESP32/nimble/BLEManagerImpl.cpp:231]: ble obj context ?

**Comment by dhrishi** [comment]
@sayondeep Please update the PR description about the expected behavior with this feature and the things you've tested. Thanks!

File: src/platform/ESP32/nimble/BLEManagerImpl.cpp (hunk 5 above)
Comment (sayondeep) [reply src/platform/ESP32/nimble/BLEManagerImpl.cpp]: made the change

File: src/platform/ESP32/BLEManagerImpl.h (hunk 4 above)
Comment (sayondeep) [reply src/platform/ESP32/BLEManagerImpl.h]: changed

File: config/esp32/components/chip/Kconfig (hunk 3 above)
Comment (sayondeep) [reply config/esp32/components/chip/Kconfig]: made the requested change

File: config/esp32/components/chip/Kconfig (hunk 2 above)
Comment (sayondeep) [reply config/esp32/components/chip/Kconfig]: made the requested change

File: config/esp32/components/chip/Kconfig (hunk 1 above)
Comment (sayondeep) [reply config/esp32/components/chip/Kconfig]: Yes, I have made the requested changes.

**Comment by InfiniteYuan** [comment]
LGTM

**Review by arkq — APPROVED** [review]

File: config/esp32/components/chip/Kconfig [hunk 7]
```diff
@@ -1200,4 +1200,13 @@ menu "CHIP Device Layer"
 
     endmenu
 
+    menu "Enable BLE Extended Announcement"
+        config ENABLE_BLE_EXT_ANNOUNCEMENT
+            bool "Enable BLE Extended Announcement"
+            default n
+            help
+                Enable BLE Extended Announcement.
```
Comment (dhrishi) [inline config/esp32/components/chip/Kconfig:1208]: Please update the help section as per our offline discussion

File: config/esp32/components/chip/Kconfig [hunk 8]
```diff
@@ -1176,8 +1176,8 @@ menu "CHIP Device Layer"
     menu "Commissioning Window Options"
         config CHIP_DISCOVERY_TIMEOUT_SECS
             int "Commissioning Window Timeout in seconds"
-            range 180 900 
-            default 900 
+            range 180 172800
```
Comment (dhrishi) [inline config/esp32/components/chip/Kconfig:1179]: This range should be possible only when the extended announcement is enabled

Comment (sayondeep) [reply config/esp32/components/chip/Kconfig]: made the change

File: config/esp32/components/chip/Kconfig (hunk 7 above)
Comment (sayondeep) [reply config/esp32/components/chip/Kconfig]: done

**Review by andy31415 — APPROVED** [review]

**Comment by dhrishi** [comment]
@sayondeep Please create a backport PR on this branch https://github.com/espressif/connectedhomeip/tree/v1.3-branch

**Pull request merged**